src/dev/build/tasks/bin/scripts/health-gateway

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+SCRIPT=$0
+
+# SCRIPT may be an arbitrarily deep series of symlinks. Loop until we have the concrete path.
+while [ -h "$SCRIPT" ] ; do
+  ls=$(ls -ld "$SCRIPT")
+  # Drop everything prior to ->
+  link=$(expr "$ls" : '.*-> \(.*\)$')
+  if expr "$link" : '/.*' > /dev/null; then
+    SCRIPT="$link"
+  else
+    SCRIPT=$(dirname "$SCRIPT")/"$link"
+  fi
+done
+
+DIR="$(dirname "${SCRIPT}")/.."
+CONFIG_DIR=${KBN_PATH_CONF:-"$DIR/config"}
+NODE="${DIR}/node/bin/node"
+test -x "$NODE"
+if [ ! -x "$NODE" ]; then
+  echo "unable to find usable node.js executable."
+  exit 1
+fi
+
+if [ -f "${CONFIG_DIR}/node.options" ]; then
+  KBN_NODE_OPTS="$(grep -v ^# < ${CONFIG_DIR}/node.options | xargs)"
+fi
+
+NODE_OPTIONS="$KBN_NODE_OPTS $NODE_OPTIONS" NODE_ENV=production "${NODE}" "${DIR}/src/cli_health_gateway/dist" "$@"
```

Comment:
@jbudz @tylersmalley Did I do this right?

Reply:
overall looks good.  noting the `not production ready` comment so I don't want to over complicate things, feel free to interpret however. 

- convention for our sh entrypoints is usually to prefix with the stack product, e.g. bin/kibana-health-gateway
- bin/health-gateway on the distrbution exits 130 atm with no logs, is that expected?  i can look into it more.
-  if this extends beyond cloud:
    - it would be helpful to commit gateway.yml so we can define user and group permissions for deb and rpm distributions.  technically we can workaround it, but it makes life easier to strictly define configuration files.
    - we may want to add a service definition to support `service kibana-health-gateway start` and put logs in the correct location, etc.
    - we may want node.options to be separate from kibana  e.g. 256mb heap health gateway, 2gb heap kibana.

not required but simplifies process monitoring:
```suggestion
NODE_OPTIONS="$KBN_NODE_OPTS $NODE_OPTIONS" NODE_ENV=production exec "${NODE}" "${DIR}/src/cli_health_gateway/dist" "$@"
```

Reply:
Thanks for the feedback! Updated and will test `bin/kibana-health-gateway` on the next build to see if I'm running into the same error you are.

> if this extends beyond cloud:

The current plan is that this will not extend beyond cloud; it'll just be used internally.

Reply:
@jbudz I pulled down the latest image, and running `./bin/kibana-health-gateway` hangs for a few seconds and then exits. Guessing there's something I did wrong in the build or cli scripts here... they were mostly copied from ones. I can't even get the `--help` text to run.

Reply:
> Guessing there's something I did wrong in the build or cli scripts here

Forgot to add `program.parse()`, which seems to have fixed it